src/main/java/com/ottofernan/librarycrud/controllers/BookRestController.java

```
@@ -17,38 +20,36 @@ public BookRestController(BookService bookService) {
     }
 
     @GetMapping("/get")
-    public Set<Book> getAll(){
+    public Set<BookDTO> getAll(){
 
         return bookService.findAll();
     }
 
     @GetMapping("/get/{id}")
-    public Book getById(@PathVariable("id") Long id){
+    public BookDTO getById(@PathVariable("id") Long id){
         return bookService.findById(id);
     }
 
     @GetMapping("/getByTitle")
-    public Set<Book> getByTitle(@ModelAttribute("title") String title){
+    public Set<BookDTO> getByTitle(@ModelAttribute("title") String title){
+        if(title == null) return null;
 
-        if(title != null){
-            return bookService.findAllByTitle(title);
-        } else {
-            return null;
-        }
+        return bookService.findAllByTitle(title);
     }
 
     @PostMapping("/post")
-    public Book postBook(@RequestBody Book book){
-        if(Book.isValid(book)){
+    public BookDTO postBook(@RequestBody BookDTO book){
+        if(Book.isValid(toModel(book))){
             bookService.save(book);
+            return book;
         }
-        return book;
+        return null; // TODO: retornar uma ResponseEntity
     }
 
     @PutMapping(value = "/post", consumes = "application/json")
-    public void updateBook(@RequestBody Book book){
+    public void updateBook(@RequestBody BookDTO book){
         if(book != null){
```

Aqui dá pra usar o fail fast tipo ```if(book == null || checkedBook == null ) return```